tests/aws/services/cloudformation/resources/test_opensearch.py

```diff
@@ -48,3 +48,18 @@ def test_domain(deploy_cfn_template, aws_client, snapshot):
     tags_result = aws_client.opensearch.list_tags(ARN=domain_arn)
     tags_result["TagList"].sort(key=itemgetter("Key"))
     snapshot.match("list_tags", tags_result)
+
+
+@markers.aws.validated
+def test_domain_with_alternative_types(deploy_cfn_template, aws_client):
+    """
+    Test that the alternative types for the OpenSearch domain are accepted using the resource documentation example
+    """
+    stack = deploy_cfn_template(
+        template_path=os.path.join(
+            os.path.dirname(__file__), "../../../templates/opensearch_domain_alternative_types.yml"
+        )
+    )
+    domain_name = stack.outputs["SearchDomain"]
+    domain = aws_client.opensearch.describe_domain(DomainName=domain_name)
+    assert domain["DomainStatus"]["DomainName"] == domain_name
```

Comment:
This should probably also test the added access policies on the API-level.
Please add snapshots here. :pray: